src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -716,11 +716,16 @@ private String retrieveGCPSnapshotRegion(SnapshotCacheResult cachedSnapshot, FSF
     if (cachedSnapshot.isSelfHosted) {
       // Authorize using the dataset's service account...
       Storage storage = gcsProjectFactory.getStorage(cachedSnapshot.datasetProjectId);
-      Bucket bucket =
-          storage.get(
-              GcsUriUtils.parseBlobUri(fsFile.getCloudPath()).getBucket(),
-              // ...but bill to the snapshot's project
-              BucketGetOption.userProject(cachedSnapshot.googleProjectId));
+      // ...but bill to the snapshot's project.
+      BucketGetOption userProject = BucketGetOption.userProject(cachedSnapshot.googleProjectId);
```

Comment:
Looking at the constructor for `SnapshotCacheResult`, it looks possible to have a null `googleProjectId` even if `isSelfHosted` is true. If that happens, I think `BucketGetOption.userProject()` will throw an NPE.

If the code is correct -- when `isSelfHosted` is true, `googleProjectId` is not null -- you can make it explicit in the code by using `requireNonNull()`
```suggestion
      BucketGetOption userProject = BucketGetOption.userProject(Objects.requireNonNull(cachedSnapshot.googleProjectId));
```

Reply:
Good shout -- I took a closer look at this.  The guarantee we can rely on is as follows:

_When a snapshot's backing cloud platform is GCP, `googleProjectId` is not null.  The snapshot's self-hosted status has no bearing on `googleProjectId`'s value._

We have a shaky chain of method definitions to get here: this code snippet lives in a method called `retrieveGCPSnapshotRegion` which should only be called for a GCP-backed snapshot.  I am inclined to expand docstrings and comments to make the chain of guarantee more clear, and save any further refactoring for the future team.  We don't currently use `Objects.requireNonNull` in production code, and it looks like in this case it would be a targeted bandaid in place of a refactor to platform-specific snapshot and/or cached snapshots.